File: Module 19-Sorting&algo/04_Selectionsort.py
arr = [23, 24, 12, 10, 22]
n = len(arr)

for i in range(n - 1):
    min_index = i
    for j in range(i + 1, n):
        if arr[j] < arr[min_index]:
            min_index = j
    if min_index != i:  # swap only if needed
        arr[i], arr[min_index] = arr[min_index], arr[i]

# Print sorted array
for num in arr:
    print(num, end=" ")
print()

File: Module 19-Sorting&algo/04_Selectionsort.cpp
#include<iostream>
#include<vector>
using namespace std;

int main(){
    vector<int> arr = {23, 24, 12, 10, 22};
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        int temp = i;
        for (int x = i; x < n ; x++) {
            if (arr[temp]>arr[x]) {
                temp = x;
            }
        }
        swap(arr[i], arr[temp]); 
    }

    // Print sorted array
    for (int num : arr) {
        cout << num << " ";
    }
    cout << endl;

    return 0;

}